### PR TITLE
Fix some ldap related issues

### PR DESCRIPTION
**Pull request by CarlSchwan**
This included three commits, see individual commit message for more in-depth explaination

1. More type hinting and typo fixing
2. Fix loading user lists
3. Another user list loading fixing

### CODE REVIEW

File: apps/user_ldap/lib/Access.php [hunk 1]
```diff
@@ -528,7 +528,7 @@ public function dn2username($fdn, $ldapName = null) {
 	 */
 	public function dn2ocname($fdn, $ldapName = null, $isUser = true, &$newlyMapped = null, array $record = null) {
 		static $intermediates = [];
-		if (isset($intermediates[$isUser ? 'user-' : 'group-' . $fdn])) {
+		if (isset($intermediates[$isUser ? 'user-' . $fdn : 'group-' . $fdn])) {
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Access.php:531]:
```suggestion
		if (isset($intermediates[($isUser ? 'user-' : 'group-') . $fdn])) {
```
Or make it more explicit?

Comment (CarlSchwan) [reply apps/user_ldap/lib/Access.php]: Done :)

File: apps/user_ldap/lib/Group_LDAP.php [hunk 2]
```diff
@@ -523,8 +528,7 @@ private function prepareFilterForUsersHasGidNumber(string $groupDN, string $sear
 	}
 
 	/**
-	 * returns a list of users that have the given group as gid number
-	 *
+	 * @return A list of users that have the given group as gid number
```
Comment (come-nc) [inline apps/user_ldap/lib/Group_LDAP.php:531]: @return should contain a type, even if mixed

**Comment by come-nc** [comment]
Apart from the docblock remark, everything in there seems to improve the situation.

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
(Lots of psalm error through, please look into them as you added typing they may unravel real problems)

**Comment by CarlSchwan** [comment]
The psaml errors are unfortunately normal in stable20 and earlier. There are none in the master equivalent of this: https://github.com/nextcloud/server/pull/30223

File: apps/user_ldap/lib/Group_LDAP.php [hunk 3]
```diff
@@ -442,10 +450,11 @@ public function gidNumber2Name(string $gid, string $dn) {
 	}
 
 	/**
+	 * @return ?string The name of the group
 	 * @throws ServerNotAvailableException
 	 * @throws Exception
 	 */
-	private function getNameOfGroup(string $filter, string $cacheKey) {
+	private function getNameOfGroup(string $filter, string $cacheKey): ?string {
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Group_LDAP.php:457]: apps/user_ldap/lib/Group_LDAP.php#L472

The declared return type 'null|string' for OCA\User_LDAP\Group_LDAP::getNameOfGroup does not allow false, but the function returns 'false|string'

**Pull request closed without merging**